modules/ting_field_search/ting_field_search.module

```diff
@@ -618,29 +782,33 @@ function ting_field_search_profiles_load($name) {
 /**
  * Loads all profiles from code or DB.
  *
- * @param bool $reset (optional)
- *   Wether or not to reset to the ctools object cache.
+ * @param bool $reset
+ *   Whether or not to reset to the ctools object cache.
  *
- * @return array $profiles.
+ * @return array
  *   An array of profile objects keyed bt profile machine name.
  */
 function ting_field_search_profiles_load_all($reset = FALSE) {
-  return ting_field_search_profiles_load_mutliple();
+  $profiles = &drupal_static(__FUNCTION__);
+  if (!isset($profiles) || $reset) {
+    $profiles = ting_field_search_profiles_load_mutliple();
```

Comment:
Typo in `mutliple` but perhaps this matters less here.

Reply:
Fixed